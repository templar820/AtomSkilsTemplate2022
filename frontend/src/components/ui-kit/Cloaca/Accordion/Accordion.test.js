import React from 'react';
import ReactDOM from 'react-dom';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import { Accordion, Collapse } from './';

Enzyme.configure({ adapter: new Adapter() });

const props = {
    isOpen: true,
    selectedItem: 1,
    closeAll: false
};

describe('<Accordion />', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(
            <Accordion {...props}>
                <Collapse />
                <Collapse />
            </Accordion>
        );
    });

    afterEach(() => {
        wrapper.unmount();
    });

    test('should match snapshot', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Accordion {...props} />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    test('should render correctly', () => {
        expect(props).toHaveProperty('isOpen', props.isOpen);
        expect(props).toHaveProperty('selectedItem', props.selectedItem);
        expect(props).toHaveProperty('children', props.children);
        expect(props).toHaveProperty('closeAll', props.closeAll);
        expect(wrapper.prop('size')).toEqual(props.size);
    });

    test('should component expand', () => {
        expect(
            wrapper
                .find('withColors(Collapse)')
                .at(props.selectedItem)
                .prop('expanded')
        ).toBeTruthy();
    });

    test('should component onClick expand', () => {
        wrapper
            .find('withColors(Collapse)')
            .at(0)
            .simulate('click');
        expect(
            wrapper
                .find('withColors(Collapse)')
                .at(props.selectedItem)
                .prop('expanded')
        ).not.toBeTruthy();
    });
});
