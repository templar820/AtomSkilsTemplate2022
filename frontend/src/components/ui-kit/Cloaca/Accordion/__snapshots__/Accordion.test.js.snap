// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accordion /> should match snapshot 1`] = `
<StyledAccordionWrapper
  className="CLR-accordion"
  closeAll={false}
  isOpen={true}
  selectedItem={1}
>
  <withColors(Collapse)
    childKey={0}
    disabled={false}
    expanded={false}
    isLoading={false}
    isOpen={true}
    key=".0"
    onClick={[Function]}
    onExpand={[Function]}
    styles={Object {}}
  />
  <withColors(Collapse)
    childKey={1}
    disabled={false}
    expanded={true}
    isLoading={false}
    isOpen={true}
    key=".1"
    onClick={[Function]}
    onExpand={[Function]}
    styles={Object {}}
  />
</StyledAccordionWrapper>
`;
